pkg/util/net.go

```diff
@@ -256,3 +256,26 @@ func IsIPv6OnlyCIDRs(cidrs []*net.IPNet) (bool, error) {
 
 	return !v4Found && v6Found, nil
 }
+
+// IPToIPNet converts an IP to an IPNet, using a fully filled mask appropriate for the address family.
+func IPToIPNet(ip net.IP) (*net.IPNet, error) {
```

Comment:
This might be a good place for a generic now that we are on go 1.18.

Reply:
yeah, I'm kind of waiting to see when Kubernetes starts using them. I think they'll probably wait until all supported branches are on 1.18 or newer, otherwise backports to older branches that aren't on 1.18 yet will be impossible.